### PR TITLE
Collect Interpreter Statistics

### PR DESCRIPTION
**Pull request by JWZ2018**
Issue: #934

Summary:
Gather and display interpreter statistics:
- objects created

Test Plan:
Ran prepack with --logStatistics and saw new statistics logged accordingly
Checked tests passed

### CODE REVIEW

File: src/serializer/serializer.js [hunk 1]
```diff
@@ -63,10 +64,19 @@ export class Serializer {
       let forbiddenNames = realmPreludeGenerator.nameGenerator.forbiddenNames;
       traverseFast(ast, node => {
         if (!t.isIdentifier(node)) return false;
-
         forbiddenNames.add(((node: any): BabelNodeIdentifier).name);
         return true;
       });
+      traverseFast(ast, node => {
```
Comment (NTillmann) [inline src/serializer/serializer.js]: By doing a walk over the AST in this way, you are counting what's statically present in the code. The intention of the referenced issue was to count which actually happens, and how often, while running the code in our the interpreter. Look at src/environment.js, there's a `evaluateAbstract` function which gets called for each AST node.

**Comment by NTillmann** [comment]
Counting each node type contains a lot of information.
To do what the issue asks for, ...
- you could compute aggregates for object creations, aggregating counts of `ArrayExpression`, `NewExpression`, or you could go into the `ObjectValue` constructor and increment a counter there.
- counting property updates via node counting is more tricky. It's probably easier to go into a helper function such as `InternalUpdatedProperty` and increment a counter there.
- To do what this comment asks for: `When forking and merging control-flow, compute the maximum value for each explored case.`, you'd probably have to add the counters to something that gets tracked by the `Effects`, and then the `join` operation computes maximums. Please discuss this if it turns out to be too much work, or not clear.

File: src/serializer/serializer.js (hunk 1 above)
Comment (JWZ2018) [reply src/serializer/serializer.js]: Ok I will relook at this and try again.

**Comment by JWZ2018** [comment]
Currently only added object creation count. Will work on the other ones later.

**Comment by JWZ2018** [comment]
Too far behind master. Closing for now, will open another one for this later.

**Pull request closed without merging**